superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx

```diff
@@ -0,0 +1,162 @@
+  ...mockedProps,
+  editMode: false,
+};
+
+fetchMock.get('glob:*/csstemplateasyncmodelview/api/read', {});
+
+async function openDropdown() {
+  const btn = screen.getByRole('button');
+  userEvent.click(btn);
+  expect(await screen.findByRole('menu')).toBeInTheDocument();
+}
+
+test('should render', () => {
+  const { container } = render(<HeaderActionsDropdown {...mockedProps} />);
+  expect(container).toBeInTheDocument();
+});
+
+test('should render the dropdown button', () => {
+  render(<HeaderActionsDropdown {...mockedProps} />);
+  expect(screen.getByRole('button')).toBeInTheDocument();
+});
+
+test('should render the dropdown icon', () => {
+  render(<HeaderActionsDropdown {...mockedProps} />);
+  expect(screen.getByRole('img', { name: 'more-horiz' })).toBeInTheDocument();
+});
+
+test('should open the dropdown', async () => {
+  render(<HeaderActionsDropdown {...mockedProps} />);
+  await openDropdown();
+  expect(await screen.findByRole('menu')).toBeInTheDocument();
+});
+
+test('should render the menu items in edit mode', async () => {
+  render(<HeaderActionsDropdown {...mockedProps} />);
+  await openDropdown();
+  expect(screen.getAllByRole('menuitem')).toHaveLength(8);
+  expect(screen.getByText('Save as')).toBeInTheDocument();
+  expect(screen.getByText('Copy dashboard URL')).toBeInTheDocument();
+  expect(screen.getByText('Share dashboard by email')).toBeInTheDocument();
+  expect(screen.getByText('Refresh dashboard')).toBeInTheDocument();
+  expect(screen.getByText('Set auto-refresh interval')).toBeInTheDocument();
+  expect(screen.getByText('Set filter mapping')).toBeInTheDocument();
+  expect(screen.getByText('Edit dashboard properties')).toBeInTheDocument();
+  expect(screen.getByText('Edit CSS')).toBeInTheDocument();
+});
+
+test('should render the menu items when not in edit mode', async () => {
+  render(<HeaderActionsDropdown {...editModeOffProps} />);
+  await openDropdown();
+  expect(screen.getAllByRole('menuitem')).toHaveLength(7);
+  expect(screen.getByText('Save as')).toBeInTheDocument();
+  expect(screen.getByText('Copy dashboard URL')).toBeInTheDocument();
+  expect(screen.getByText('Share dashboard by email')).toBeInTheDocument();
+  expect(screen.getByText('Refresh dashboard')).toBeInTheDocument();
+  expect(screen.getByText('Set auto-refresh interval')).toBeInTheDocument();
+  expect(screen.getByText('Download as image')).toBeInTheDocument();
+  expect(screen.getByText('Toggle fullscreen')).toBeInTheDocument();
+});
+
+test('should render the "Save Modal" when user can save', async () => {
+  render(<HeaderActionsDropdown {...mockedProps} />);
+  await openDropdown();
+  expect(screen.getByText('Save as')).toBeInTheDocument();
+});
+
+test('should NOT render the "Save Modal" menu item when user cannot save', async () => {
+  const cannotSaveProps = {
+    ...mockedProps,
+    userCanSave: false,
+  };
+  render(<HeaderActionsDropdown {...cannotSaveProps} />);
+  await openDropdown();
```

Comment:
Same thing here. First I guarantee that it exists and then I see that it no longer exists.

Reply:
I think here as well you are referring to the previous test `should render the "Save Modal" when user can save` which is testing the opposite case where the "Save Modal" does exist.